src/pallene/ir.lua

```diff
@@ -186,15 +194,14 @@ local ir_cmd_constructors = {
     --
     -- Control flow
     --
-    Nop     = {},
-    Seq     = {"cmds"},
+    InitFor    = {"loc", "dst_i", "dst_cond", "dst_iter", "dst_count",
+                  "src_start", "src_limit", "src_step"},
 
-    Return  = {"loc", "srcs"},
-    Break   = {},
-    Loop    = {"body"},
+    IterFor    = {"loc", "dst_i", "dst_cond", "dst_iter", "dst_count",
+                  "src_start", "src_limit", "src_step"},
 
-    If      = {"loc", "src_condition", "then_", "else_"},
-    For     = {"loc", "dst", "src_start", "src_limit", "src_step", "body"},
+    Jmp        = {"target"},
+    JmpIfFalse = {"loc", "src_cond", "target"},
```

Comment:
Why does one have a loc and the other not?

Reply:
Jmp used to have a loc too, but stripping it off made the i.r. generation simpler. It was easier to add an indirect jmp without having to reference any a.s.t. node in particular. JmpIfFalse needs a loc because of "src_cond". If there's an error related to the source (e.g. it not being initialized), we need it's loc.